clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2352,26 +2352,61 @@ static void handleUnusedAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   D->addAttr(::new (S.Context) UnusedAttr(S.Context, AL));
 }
 
-static void handleConstructorAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
-  uint32_t priority = ConstructorAttr::DefaultPriority;
+static bool diagnoseInvalidPriority(Sema &S, uint32_t Priority,
+                                    const ParsedAttr &A,
+                                    SourceLocation PriorityLoc) {
+  // Only perform the priority check if the attribute is outside of a system
+  // header. Values <= 100 are reserved for the implementation, and libc++
+  // benefits from being able to specify values in that range.
+  if ((Priority < 101 || Priority > 65535) &&
+      !S.getSourceManager().isInSystemHeader(A.getLoc())) {
```

Comment:
Going from the comment, shouldn't it be 

```cpp
(Priority < 101 && !S.getSourceManager().isInSystemHeader(A.getLoc())) || Priority > 65535)
```

Reply:
Ah, no, I can clarify the comment -- the 65535 is the documented upper bound, but we could reasonably extend that in the future if we wanted.